Source/WTF/wtf/Compiler.h

```diff
@@ -447,25 +447,33 @@
     _Pragma(_COMPILER_STRINGIZE(compiler diagnostic push)) \
     _COMPILER_CONCAT(IGNORE_WARNINGS_BEGIN_IMPL_, cond)(compiler, warning)
 
+/* condition is either 1 (true) or 0 (false, do nothing) */
 #define IGNORE_WARNINGS_BEGIN_IMPL_1(compiler, warning) \
     _Pragma(_COMPILER_STRINGIZE(compiler diagnostic ignored warning))
 #define IGNORE_WARNINGS_BEGIN_IMPL_0(compiler, warning)
-#define IGNORE_WARNINGS_BEGIN_IMPL_(compiler, warning)
-
 
+#if defined(__has_warning)
 #define IGNORE_WARNINGS_END_IMPL(compiler) _Pragma(_COMPILER_STRINGIZE(compiler diagnostic pop))
+#else
+#define IGNORE_WARNINGS_END_IMPL(compiler) \
+    _Pragma(_COMPILER_STRINGIZE(compiler diagnostic pop)) \
+    _Pragma(_COMPILER_STRINGIZE(compiler diagnostic pop))
+#endif
 
 #if defined(__has_warning)
 #define _IGNORE_WARNINGS_BEGIN_IMPL(compiler, warning) \
     IGNORE_WARNINGS_BEGIN_COND(__has_warning(warning), compiler, warning)
 #else
-#define _IGNORE_WARNINGS_BEGIN_IMPL(compiler, warning) IGNORE_WARNINGS_BEGIN_COND(1, compiler, warning)
+/* Suppress -Wpragmas to dodge warnings about attempts to suppress unrecognized warnings. */
```

Comment:
Does this have any other costs? For example, will we lose warnings about non-warning-suppressing pragmas? Maybe only between the IGNORE_WARNINGS_BEGIN/END pair.

Reply:
Yes, between the IGNORE_WARNINGS_BEGIN/END, and only for GCC. This problem will go away automatically in the future if GCC decides to support __has_warning(), like Clang does.